includes/api/class-wc-rest-authentication.php

```diff
@@ -50,19 +50,12 @@ public function __construct() {
 	 * @return bool
 	 */
 	protected function is_request_to_rest_api() {
-		if ( empty( $_SERVER['REQUEST_URI'] ) ) {
-			return false;
-		}
-
 		$rest_prefix = trailingslashit( rest_get_url_prefix() );
 
-		// Check if our endpoint.
-		$woocommerce = ( false !== strpos( $_SERVER['REQUEST_URI'], $rest_prefix . 'wc/' ) ); // @codingStandardsIgnoreLine
-
 		// Allow third party plugins use our authentication methods.
 		$third_party = ( false !== strpos( $_SERVER['REQUEST_URI'], $rest_prefix . 'wc-' ) ); // @codingStandardsIgnoreLine
```

Comment:
Since we still are checking for `$_SERVER['REQUEST_URI']` we should keep the check if is empty.
Also should we allow `wc-` inside `WC()->is_rest_api_request()`? Should be nice for some third-party plugins.

Reply:
Of course you're right on the first point, thanks for noticing. 

Ad the second point, I assume plugins can filter `woocommerce_is_rest_api_request` if they want to?